dev/com.ibm.ws.microprofile.config.1.1.cdi/src/com/ibm/ws/microprofile/config/cdi/ConfigProducer.java

```diff
@@ -33,22 +35,43 @@
 
     private static final TraceComponent tc = Tr.register(ConfigProducer.class);
 
-    public static Object newValue(Config config, InjectionPoint injectionPoint, Type type, boolean optional) {
+    public static Object newValue(Config config, InjectionPoint injectionPoint) {
+
+        Type ipType = injectionPoint.getType();
         ConfigProperty qualifier = getConfigPropertyAnnotation(injectionPoint);
         String defaultValue = qualifier.defaultValue();
         String propertyName = getPropertyName(injectionPoint, qualifier);
-        Object value = newValue(config, propertyName, defaultValue, type, optional);
+
+        Object value = newValue(config, propertyName, defaultValue, ipType);
+
         return value;
     }
 
-    public static Object newValue(Config config, String propertyName, String defaultValue, Type type, boolean optional) {
+    public static Object newValue(Config config, String propertyName, String defaultValue, Type type) {
```

Comment:
The methods in this class could really have done with some comments to explain what they're expecting and which parts of the config lookup process they handle.

Looking at this PR, it looks like there are several similar-looking methods which each handle different parts of the process and call into each other.